### DISCUSSION TITLE
way to customize autocomplete dropdown click handling?

### DISCUSSION THREAD

**Discussion by orichh** [mantinedev/mantine#2254] [Q&A] [answered]
hi everyone - new to mantine and loving it so far!

i'm trying to customize the autocomplete dropdown and having trouble. i was able to render buttons on the custom autocomplete item component but every time i click on one of the buttons, it doesn't do anything except close the dropdown.

<img width="369" alt="image" src="https://user-images.githubusercontent.com/18966944/187043733-e599bf51-3784-477c-b9c9-c3824d2ebadc.png">

i was able to go into the devtools and found that if i remove the focusOut event listener on the div with class name "mantine-Autocomplete-dropdown" and then remove the mouseDown event listener from that div's child div, that i can get it to stay open. however, i cannot find a way to actually do this outside of the devtools.

<img width="396" alt="image" src="https://user-images.githubusercontent.com/18966944/187043856-dcc2d849-7d2b-496e-9e00-7a3764066c6d.png">

is there a way i can customize the opening and closing behavior for this component?

apologies if this has been answered elsewhere!

**Comment by rtivital** [comment, marked as answer]
Hi @richardhyesungo, Autocomplete and other similar components implement WAI-ARIA accessibility recommendations – it is not allowed to use extra controls in items as their only purpose is to provide suggestions for filling the input. I recommend using Popover component for this instead.

**Reply by orichh** [reply]
@rtivital ok, thank you!
